src/v1/middlewares/validators/validator.ts

```diff
@@ -0,0 +1,34 @@
+// validateResource.ts
+import { Request, Response, NextFunction } from 'express';
+import { z, AnyZodObject } from 'zod';
+import { responseStatusCodes } from '../../helpers/common';
+import {
+  VALIDATION_ERROR,
+  INTERNAL_SERVER_ERROR,
+} from '../../helpers/locale.json';
+
+const validateResource =
+  (schema: AnyZodObject) =>
+  (req: Request, res: Response, next: NextFunction) => {
+    try {
+      schema.parse({
```

Comment:
How will it know what to validate against the schema, out of these three?

Reply:
Because in the schema we define its type, such as body, query, or params, it will automatically pick that up, which is the default behavior of Zod with this method.